### ISSUE TITLE
[BUG] when making a 'module' node, typein the same name of the module iteself will cause the page freeze.

### ISSUE THREAD

**Issue by lancety** [retejs/rete.js.org#45]
step to reproduce
* open https://rete.js.org/#/examples/module#lang=en&tosearch=%20separate%20plugin%20for%20those%20who%20need
* clean the input of "Module" node in index.rete
* type in "index"
* when typing "x" letter, the page will freeze.

**Comment by Ni55aN** [comment]
indeed, an additional check is needed to prevent the module from being used recursively
